### PR TITLE
chore: bump cert-manager to v0.2.1

### PR DESCRIPTION
**Pull request by jr0d**
**What type of PR is this?**

Chore

**What this PR does/ why we need it**:

Updates cert-manager-setup to v0.2.1 and cert-manager to v1.0.1. This is a breaking change and requires all dependent addons to be updated to use the cert-manager.io/v1 API.

**Which issue(s) this PR fixes**:

https://jira.d2iq.com/browse/D2IQ-70659

**Special notes for your reviewer**:
This is a breaking change.

**Does this PR introduce a user-facing change?**:

```release-note
cert-manager:
    * v1 API
    * Renaming our API group from certmanager.k8s.io to cert-manager.io
    * Removal of the v1alpha API
    * kubectl cert-manager status command to help with investigating issues
    * Using new and stable Kubernetes APIs
    * Improved logging
    * ACME improvements
    * kubectl cert-manager create certificaterequest for signing local certificates
    * General Availability of JKS and PKCS#12 keystore support
    * kubectl cert-manager CLI plugin allowing manual renewal and API version conversion
    * ACME External Account Binding support
    * Support for full set of x509 ‘subject’ parameters
```

**Checklist**

* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Review by joejulian — APPROVED** [review]

**Review by dkoshkin — APPROVED** [review]

**Comment by joejulian** [comment]
```
Reason:DriverDeployFailure,Message:chart "cert-manager-setup" version "0.2.1" not found in https://mesosphere.github.io/charts/staging repository
```

File: addons/cert-manager/cert-manager.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
   chartReference:
     chart: cert-manager-setup
     repo: https://mesosphere.github.io/charts/staging
-    version: 0.1.17
+    version: 0.2.1
```
Comment (gracedo) [inline addons/cert-manager/cert-manager.yaml:30]:
```suggestion
    version: 0.2.0
```
I think the latest version is [`0.2.0`](https://github.com/mesosphere/charts/blob/master/staging/cert-manager-setup/Chart.yaml#L4)

Comment (gracedo) [reply addons/cert-manager/cert-manager.yaml:30]: (nvm on this since we've since published a newer version `0.2.1`)

**Pull request merged**